crates/router/src/routes/dummy_connector/utils.rs

```diff
@@ -15,8 +15,11 @@ use crate::{configs::settings, routes::AppState};
 
 pub async fn tokio_mock_sleep(delay: u64, tolerance: u64) {
     let mut rng = rand::thread_rng();
-    let effective_delay = rng.gen_range((delay - tolerance)..(delay + tolerance));
-    tokio::sleep(tokio::Duration::from_millis(effective_delay)).await
+    let effective_delay = Uniform::from((delay - tolerance)..(delay + tolerance));
```

Comment:
can we use `Uniform::try_from` ?.

(although that also does not seem to help)
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4a99d8a5b9a9604292cf7065e83e2866

Maybe it'll be fixed in future, we can use the safer API on our side in the meanwhile

Reply:
@lsampras  I started with `try_from`:
`    let timeout = Uniform::try_from(0..=settings.loop_interval)?;`
But the compiler complains that:

```bash
the trait `std::convert::From<Infallible>` is not implemented for `error_stack::Report<ProcessTrackerError>`
```

We can't just implement `From<Infallible>` for `error_stack::Report<ProcessTrackerError>` because of Rusts orphan rule. If we want to get around this then we may have to add significant amount of workaround in the `errors.rs` file potentially changing the API contract. 
Plus as you can see above it says to implement something that is Infallible that is unlikely to fail so we are not gaining much here. And another reason is that the `from` method just does type conversion here which is infallible; the actual random number work is done by the `sample` method which again doesn't return a `Result`.

Does that reasoning make sense?

Reply:
@lsampras @SanchithHegde any thoughts on the above comment?

Reply:
```
let timeout = Uniform::try_from(0..=settings.loop_interval).into_report().change_context(ProcessTrackerError::InvalidInput)?;
```
I believe this should fix it^

yeah the result is currently infallible but in the future it's not gonna be,
we won't have much benefit from adding it rn but we'll be benefitting from this when a new version of rand is released

Reply:
@lsampras cool, thanks. That is good to know. I'll resubmit the PR.

Reply:
I believe this still needs to be changed to `Uniform::try_from()`?